src/main/java/seedu/address/logic/parser/SetStatusCommandParser.java

```diff
@@ -0,0 +1,44 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.SetStatusCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Person;
+import seedu.address.model.project.Task;
+
+
+/**
+ * Parses input arguments and creates a new AddCommand object
+ */
+public class SetStatusCommandParser implements Parser<SetStatusCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddCommand
+     * and returns an AddCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SetStatusCommand parse(String args) throws ParseException {
+        String status = args.split(" of")[0].trim();
+        String taskAndProject = args.split(" of")[1].trim();
+        String taskName = taskAndProject.split("in ")[0].trim();
+        String projectName = taskAndProject.split("in ")[1];
+        if (projectName.length() == 0) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SetStatusCommand.MESSAGE_USAGE));
+        }
+        Person project = new Person(ParserUtil.parseName(projectName));
+        Task newTask = new Task(taskName);
+        return new SetStatusCommand(status, newTask, project);
```

Comment:
What if we create 2 types of constructor instead, 1 for set status of project and another ofr set status of task.